### PR TITLE
Handle Report_Text separately.

### PR DESCRIPTION
**Pull request by akwok-dfci**
Sorry this is kind of a big PR:

* Change `reader` so that breaks in Report_Text fields (supposedly OK according to RPDR spec) are handled separately from breaks in other fields.
    * Assumes Report_Text is the last given field; throws an error if it finds a field named Report_Text and it is not the last field.
* Change default behavior of `reader` to be to raise an error when a broken (non-Report_Text) row is found. User can specify if they want the function to try to piece the rows together into complete records, or if they want the function to skip the broken rows.
* Will always throw an error if it runs into a record with length greater than the length of the header.
* Add logging warning message if malformed rows are found.
* Add logging info message if no malformed rows are found.

### CODE REVIEW

File: rpdrtools/io/io.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    """Returns a generator that yields the records of a RPDR *.txt file
+    as a list of strings.
+
+    Currently, RPDR files cannot be correctly read by regular CSV tools (e.g.
+    csv.reader, pd.read_csv).
+
+    Due to some formatting quirks with line breaks/quoting, records may be
+    split up over several rows within the CSV.
+
+    If on_broken_records is True, this function attempt to piece together the
```
Comment (benfl-dfci) [inline rpdrtools/io/io.py]:
```suggestion
    If on_broken_records is "repair", this function attempts to piece together the
```

I might also add a list of what OnBrokenRecordsType can be in the docstring so the user doesn't have to look it up themselves.

File: rpdrtools/io/io.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    RPDR_NEWLINE_CHAR,
+    RPDR_FILE_DELIMITER,
+    RPDR_REPORT_TEXT_FIELD,
+    RPDR_REPORT_END_TOKEN,
+)
+from .types import OnBrokenRecordsType
+
+
+def _merge_rows(
+    row1: List[str], row2: List[str], newline_char: str = RPDR_NEWLINE_CHAR
```
Comment (benfl-dfci) [inline rpdrtools/io/io.py:18]: I think the default newline should just be "\n". In general I think we'd want to use the "\r\n" newline character as little as possible. The input files have to have it, but I don't think anyone wants the "\r" character inside their dataframes or whatever.

Comment (benfl-dfci) [reply rpdrtools/io/io.py:18]: Decided to leave it as it is. Python's csv module is similar: it preserves Windows-style carriage returns inside quoted fields.

File: tests/test_io.py [hunk 3]
```diff
@@ -137,4 +223,187 @@ def test_no_records(self, no_records: pd.DataFrame) -> None:
...
+                "Report_Description": ["ABC\r\nDE", "ABCDE"],
+                "Report_Status": ["F", "F"],
+                "Report_Type": ["GHIJK", "GHIJK"],
+                "Report_Text": [
+                    (
+                        "\r\nThis is an example report text. "
+                        "The format may not reflect a typical note."
+                        "\r\n\r\nPatient name: First Last\r\n\r\n"
+                        "Sex: Male\r\n\r\nDOB: 1/1/1950\r\n\r\n"
+                        "HPI: ?\r\n[report_end]"
```
Comment (benfl-dfci) [inline tests/test_io.py]: My preference would be to strip `[report_end]` since I see it as part of the RPDR format and not part of the text.

Comment (akwok-dfci) [reply tests/test_io.py]: Updated in most recent commit(s).

**Review by benfl-dfci — APPROVED** [review]

**Pull request merged**